### ISSUE TITLE
Ask to install if not on app page

### ISSUE THREAD

**Issue by vertigo220** [Iamlooker/Droid-ify#115]
I'm on Android 12, so my understanding was that it should be able to auto-install, but for whatever reason it always(?) pops up asking to install. Unfortunately, this means I have to sit around and wait, because if I back out of the app's page to go back to the search results and keep looking before it pops up, it never does, and the app doesn't install.

**Comment by Iamlooker** [comment]
We cannot auto-install apps on the device, the new API is for silently updating apps but that also have a caveat. In short, silent installation is not possible. It does show notification when the app is ready to install, I will try to make it such that it will show pop-up when you are in app

**Comment by vertigo220** [comment]
I should clarify this happens not just on initial install but when updating as well, where I click the update button and it downloads the update, which I have to wait for. This is because if I go back to the update list before it finishes, any number of issues arise. Sometimes it seems to completely stop the update, and I have to redownload it when I go back and click update again.

And just testing it now, on one app I clicked update and, once it was done downloading, it popped up asking if I wanted to update the app, and I said cancel, yet the button switched to say "Launch" and the app disappeared from the update list (I tried it again on another app and it did the same thing). On another app, I clicked update and, once the download started, I went back to the update list and clicked a different app and tried clicking the update button for that app, and now Droid-ify froze and remained that way for several minutes until I killed it. When this happened, it also made the recent apps switcher very slow to show, so it seems it was dragging down the whole system. And two of the apps won't update at all, they just keep downloading then doing nothing.

**Comment by Iamlooker** [comment]
This has been fixed.

The notification will be persistent until the app is installed. Also it will open the app page when clicked so It will be much easier and smooth process, but it is still not 100% perfect

**Comment by vertigo220** [comment]
This is better, but not really what I had in mind. I was hoping for an in-app pop-up when an app is done downloading asking to install it, just like you get if you stay on the app page. So if I click update on an app and wait, it downloads then asks if I want to install it, and I say yes and it installs. If I click update and go to another app page, when it finishes downloading I don't get any prompt, so I have to either go back to the app page for the app I'm updating and click update again or, now, pull down the notifications, find the one for the app, and click it to be taken back to that page. What I'd like is for it to act the same as if I had stayed on the app page. This means for apps that show the pop-up asking to install, it would show this wherever I am in Droid-ify, and for apps that don't, it would just update them.

That latter point is especially important, because I've found that the majority of apps will just auto-install once they finish downloading if I stay on the page, whereas it seems none do if I leave the page. And that issue seems like the biggest problem here. For example, I just updated a few apps, each time staying on the page after clicking update, and each time they downloaded and, after a few seconds, the "Update" button switched to say "Launch." I then updated several apps, backing out of their page once they started downloading, and every one of them I had to go back and click the "Update" button again.

**Comment by vertigo220** [comment]
Just ran into another issue related to this. Went through and installed several apps, then checked the notifications later (since that's the only place it provides any feedback) and saw a few of them downloaded but didn't install. The notifications say to click to install the app, but each one of them, when I clicked the notification, it just took me to the main page in Droid-ify, not to the app. And, of course, the notification then went away, so I couldn't check it again to see which app it was, which I didn't remember since I wasn't expecting it to do this (the first time, after that I paid closer attention), so I had to go through and redo my previous searches looking for the apps I'd installed to find which one it was in order to install it.

So there are two problems here:

1. The notifications don't work properly, i.e. they don't take you to the app's page.
2. There is no place in Droid-ify that shows downloaded but not updated/installed apps, so any apps stuck in this limbo phase without a notification (whether due to the issue mentioned here or if notifications are cleared, perhaps by a reboot or just by accident) will be left that way unless the user can remember each one of them and individually check them manually.

@Iamlooker I assumed you'd see these updates, but you haven't replied or reopened the issue so mentioning you to make sure you see them.

**Comment by Iamlooker** [comment]
Sorry but it is not currently possible to show install prompts on all pages.

**Comment by vertigo220** [comment]
I've been going through and installing a bunch of apps the past couple days, and I've had many of them show the install prompts while on other pages, both in other apps' pages and on the main page. Of course, this can be problematic since, whether still on the app's page or somewhere else, when the install prompt shows, if you click off of it (and also once when I clicked cancel), which is easy to do if you're scrolling and it suddenly pops up, it will get stuck in the waiting state and the only way to get it to install is to close and reopen Droid-ify and click install again. Hopefully this is fixed in the latest version, so once I'm able to update to it I'll test it out.

I also had at least one time where I hit install on an app, the button changed to "Cancel" (which, by the way, almost never works) but didn't show any progress happening, I went back to the main screen then back to the app page, and the button had changed back to "Install," and I was able to repeat it at least once. And another app still showed the "Installing" with the progress bar even after the button changed to "Launch." And another that got stuck on "Installing" with the "Cancel" button grayed out when I switched apps. So again hopefully these have been fixed in 0.5.7, but want to mention them now before I forget so I can check them later.
